pkg/git/cache.go

```diff
@@ -0,0 +1,141 @@
+package git
+
+import (
+	"crypto/sha1"
+	"encoding/hex"
+	"hash"
+	"os"
+	"path"
+	"sync"
+
+	"github.com/git-depend/git-depend/pkg/utils"
+)
+
+// Stores the path of the cache
+type Cache struct {
+	sync.Mutex
+	// root path to the cache
+	root string
+	// repositories maps the URL to the filepath
+	repositories map[string]string
+}
+
+func NewCache(path string) (*Cache, error) {
+	_, err := os.Stat(path)
+	if os.IsNotExist(err) {
+		if err := os.Mkdir(path, 0755); err != nil {
```

Comment:
Will this handle nested paths? MkdirAll?

Reply:
```bash
> go run git-dep.go commit
Using config file: C:\Users\co_aly\.git-depend.toml
mkdir C:\Users\co_aly/.cache/git-depend: The system cannot find the path specified.
```

Reply:
Yes I'm on windows, forgive me

Reply:
omg

Reply:
Fixed but need to logon to Windows to verify.

Reply:
Also, is it just the error message which displays the slashes the wrong way around or does it actually create them that way around on Windows?

i.e it should be:
```
C:\Users\co_aly\.cache\git-depend
```

Reply:
So, I'm using https://scoop.sh to manage my windows environments.  I then started a CMD processor, not Powershell
```
     scoop install go114
     git clone https://github.com/git-depend/git-depend.git 
     git checkout finn/dev 
     go run git-dep.go config --author="Finn Ball" --email="finn.ball@codificasolutions.com" 
     go run git-dep.go add https://github.com/git-depend/repoA.git:my/branch 
     go run git-dep.go commit   
 ```
 
 And that functions as expected. I believe.

Reply:
Recent versions of windows support / and \ slashes so it shouldn't be a problem, the comment was mainly on the fact we cant assume $home/.cache to exist

Reply:
arlyon, can you run "go test ./..." and let me know how long the test takes to execute? On Linux it is ~0.5s on my desktop.